crates/walrus-service/src/node.rs

```diff
@@ -1579,15 +1617,16 @@ impl StorageNode {
         Ok(())
     }
 
-    /// Initiates a committee transition to a new epoch.
+    /// Initiates a committee transition to a new epoch. Upon the return of this function, the
+    /// latest committee on chain is updated to the new node.
     ///
-    /// Returns `true` if epoch change event has started or was sufficiently recent such
-    /// that it should be handled.
+    /// Returns the action to execute epoch change based on the result of committee service,
+    /// including possible actions to enter recovery mode due to the node being severely lagging.
     #[tracing::instrument(skip_all)]
     async fn begin_committee_change(
```

Comment:
Remark: AFAICT, the `CommitteeService::begin_committee_change` function is only used here (+ some tests). This results in effectively doing (almost) the same case distinction twice. Maybe this could be simplified (in a separate refactoring)?